util/errors/errors.go

```diff
@@ -93,6 +94,22 @@ func isTransientEtcdErr(err error) bool {
 	return false
 }
 
+func isTransientInterruptionErr(err error) bool {
+	// Handle transient node/resource related errors
+	if strings.Contains(err.Error(), "imminent node shutdown") {
+		return true
+	} else if strings.Contains(err.Error(), "node was low on resource") {
+		return true
+	} else if strings.Contains(err.Error(), "pod deleted") {
+		return true
+	} else if strings.Contains(err.Error(), "node is shutting down") {
+		return true
+	} else if strings.Contains(err.Error(), "Deadline exceeded") {
```

Comment:
This is a reference to this error, right?
https://github.com/argoproj/argo-workflows/blob/e0a0f2856ba6e8a49b1ea28b15d50d39f3106125/workflow/controller/operator.go#L120
https://github.com/argoproj/argo-workflows/blob/e0a0f2856ba6e8a49b1ea28b15d50d39f3106125/workflow/controller/operator.go#L2007-L2012

This isn't always a transient error, and I noticed we have code that has special handling for this, e.g. https://github.com/argoproj/argo-workflows/blob/e0a0f2856ba6e8a49b1ea28b15d50d39f3106125/workflow/controller/operator.go#L374-L378

so I'm not sure this is safe.